torchao/experimental/_linear_8bit_act_xbit_weight_subclass_quantizer.py

```diff
@@ -0,0 +1,408 @@
+                packed_weight = _pack_weights_native(
+                    int_data, scale, zero_point, layout_type
+                )
+                scale = None
+                zero_point = None
+                return cls(packed_weight, scale, zero_point, layout_type)
+        except Exception as e:
+            logger.warning(
+                f"A failure occurred when packing weights with IntxWeightLayoutType.target={layout_type.target}: {e}\n"
+                + "Falling back to **slow** implementation IntxWeightLayoutType.target=fallback."
+            )
+            layout_type.target = "fallback"
+
+        # Fallback
+        assert _target_equal(layout_type.target, "fallback")
+        packed_weight = int_data.to(torch.int8)
+        return cls(packed_weight, scale, zero_point, layout_type)
+
+    def _apply_fn_to_data(self, fn):
+        self.packed_weight = fn(self.packed_weight)
+        if self.scale is not None:
+            self.scale = fn(self.scale)
+
+        if self.zero_point is not None:
+            self.zero_point = fn(self.zero_point)
+        return self
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args, kwargs):
+        kwargs = {} if kwargs is None else kwargs
+
+        if func is torch.ops.aten.detach.default:
+            return return_and_correct_aliasing(
+                func, args, kwargs, args[0]._apply_fn_to_data(torch.detach)
+            )
+        if func is torch.ops.aten.clone.default:
+            return return_and_correct_aliasing(
+                func, args, kwargs, args[0]._apply_fn_to_data(torch.clone)
+            )
+
+        raise NotImplementedError(
+            f"IntxWeightAQTLayout dispatch: attempting to run {func}, this is not supported"
+        )
+
+    def __tensor_flatten__(self):
+        if _target_equal(self.get_layout_type().target, "native"):
+            return ["packed_weight"], [self.get_layout_type()]
+
+        # fallback
+        assert _target_equal(self.get_layout_type().target, "fallback")
+        if self.zero_point is None:
+            return ["packed_weight", "scale"], [self.get_layout_type()]
+        return ["packed_weight", "scale", "zero"], [self.get_layout_type()]
+
+    @classmethod
+    def __tensor_unflatten__(
+        cls, tensor_data_dict, tensor_attributes, outer_size, outer_stride
+    ):
+        packed_weight, scale, zero_point = (
+            tensor_data_dict["packed_weight"],
+            tensor_data_dict.get("scale", None),
+            tensor_data_dict.get("zero_point", None),
+        )
+        (layout_type,) = tensor_attributes
+        return cls(packed_weight, scale, zero_point, layout_type)
+
+
+def _linear_int8_dynamic_activation_intx_weight_check(
+    input_tensor, weight_tensor, bias
+):
+    layout_type = weight_tensor.layout_tensor.layout_type
+    return isinstance(layout_type, IntxWeightLayoutType) and bias is None
+
+
+def _linear_int8_dynamic_activation_intx_weight_fallback_impl(
```

Comment:
see fallback here: https://github.com/pytorch/ao/blob/900f9acccb8bc3c102d3a7095d1b1f0355057bec/torchao/dtypes/affine_quantized_tensor.py#L1760-L1764 I think you should be able to do the same here

Reply:
Used weight_tensor.dequantize() below